src/selections/parser.cpp

```diff
@@ -149,7 +149,19 @@ Ast Parser::expression() {
 
 Ast Parser::selector() {
     if (match(Token::LPAREN)) {
-        auto ast = expression();
+        // There is an ambiguity in the grammar: an opening parenthesis can
```

Comment:
The ambiguity is resolved by this code: we first try to parse a boolean expression, and then fall back to a math expression if the previous one fails. The users should never need to worry, the only thing that this will change is the error message in case of an invalid selection string (we will get the math error message instead of the one from the boolean selection code path).